spec/models/ext_management_system_spec.rb

```diff
@@ -404,71 +404,65 @@
       expect(ExtManagementSystem.count).to eq(0)
     end
 
-    it "does not destroy an ems with active workers" do
+    it "destroys an ems with active workers" do
       ems = FactoryGirl.create(:ext_management_system)
-      FactoryGirl.create(:miq_ems_refresh_worker, :queue_name => ems.queue_name, :status => "started")
+      worker = FactoryGirl.create(:miq_ems_refresh_worker, :queue_name => ems.queue_name, :status => "started")
       ems.destroy
-      expect(ExtManagementSystem.count).to eq(1)
+      expect(ExtManagementSystem.count).to eq(0)
+      expect(worker.class.exists?(worker.id)).to be_falsy
     end
   end
 
-  context "#queue_destroy" do
+  context ".destroy_queue" do
     let(:server) { EvmSpecHelper.local_miq_server }
     let(:zone)   { server.zone }
+    let(:ems)    { FactoryGirl.create(:ext_management_system, :zone => zone) }
+    let(:ems2)    { FactoryGirl.create(:ext_management_system, :zone => zone) }
 
-    context "with no child managers" do
-      let(:ems) do
-        FactoryGirl.create(:ext_management_system, :zone => zone)
-      end
-
-      it "returns a task" do
-        task_id = ems.destroy_queue
+    it "queues up destroy" do
+      described_class.destroy_queue([ems.id, ems2.id])
 
-        deliver_queue_message
-
-        task = MiqTask.find(task_id)
-        expect(task.state).to  eq("Finished")
-        expect(task.status).to eq("Ok")
-      end
-
-      it "re-schedules with active workers" do
-        FactoryGirl.create(:miq_ems_refresh_worker, :queue_name => ems.queue_name, :status => "started", :miq_server => server)
-        ems.destroy_queue
-
-        expect(MiqQueue.count).to eq(1)
+      expect(MiqQueue.count).to eq(2)
+      expect(MiqQueue.pluck(:instance_id)).to include(ems.id, ems2.id)
```

Comment:
Shouldn't this be `match_array`?  I wouldn't expect any other `instance_id`s to be on the queue.

Reply:
right!